pandas/tests/extension/test_common.py

```diff
@@ -9,30 +12,81 @@
 from pandas.core.arrays import ExtensionArray
 
 
-class DummyDtype(dtypes.ExtensionDtype):
+class DummyClass:
     pass
 
 
+class DummyDtype(dtypes.ExtensionDtype):
+
+    type = DummyClass
+    name = "dummy"
+
+    @classmethod
+    def construct_from_string(cls, string):
+        if string == cls.name:
+            return cls()
+        else:
+            raise TypeError(f"Cannot construct a '{cls}' from '{string}'")
+
+    @classmethod
+    def construct_array_type(cls):
+        return DummyArray
+
+
 class DummyArray(ExtensionArray):
+
+    _dtype = DummyDtype
+
     def __init__(self, data):
-        self.data = data
+        self.data = np.array(data)
 
-    def __array__(self, dtype):
+    def __array__(self, dtype=None):
         return self.data
 
     @property
     def dtype(self):
-        return DummyDtype()
+        return DummyArray._dtype()
 
     def astype(self, dtype, copy=True):
         # we don't support anything but a single dtype
-        if isinstance(dtype, DummyDtype):
+        if isinstance(dtype, self._dtype):
             if copy:
                 return type(self)(self.data)
             return self
 
         return np.array(self, dtype=dtype, copy=copy)
 
+    def __len__(self):
+        return len(self.data)
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        if isinstance(scalars, cls._dtype.type):
+            scalars = [scalars]
+        return DummyArray(scalars)
+
+    def take(self, indices, allow_fill=False, fill_value=None):
```

Comment:
do we need all of the new methods this implements?

Reply:
Hi @jbrockmendel , apologies for the slow reply.  These methods, and those in the [alternative solution](https://github.com/pandas-dev/pandas/pull/44615#issuecomment-1023836417) are the minimum required for the tests to pass.